### ISSUE TITLE
dotnet nuget list source should say where a source is coming from

### ISSUE THREAD

**Issue by eerhardt** [NuGet/Home#10548]
## Details about Problem

NuGet product used (NuGet.exe | Visual Studio | MSBuild.exe | dotnet.exe):

```
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.102
 Commit:    71365b4d42

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.102\
```

Product version: 

Worked before? If so, with which NuGet version: no

## Repro steps and/or sample project

I have an old, out-of-date nuget source somewhere on my machine, but I don't know where it is. It is causing my restore to fail because the end point is no longer there.

To try to track it down, I run

```
❯ dotnet nuget list source --format Detailed
Registered Sources:
  1.  nuget.org [Enabled]
      https://api.nuget.org/v3/index.json
  2.  dotnet-tools [Enabled]
      https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
  3.  dotnet-eng [Enabled]
      https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json
  4.  dotnet5 [Enabled]
      https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json
  5.  dotnet5-transport [Enabled]
      https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5-transport/nuget/v3/index.json
  6.  dotnet6 [Enabled]
      https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json
  7.  dotnet6-transport [Enabled]
      https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6-transport/nuget/v3/index.json
  8.  richnav [Enabled]
      https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-buildservices/nuget/v3/index.json
```

It would be great if there was a format that told me where the source was coming from. Is it coming a nuget.config in the folder hierarchy? In my user settings? Where?

**Comment by JonDouglas** [comment]
@eerhardt If I understand you correctly. You would like to see the location the source originated from on disk?

i.e. 

- `C:\Users\<Name>\AppData\Roaming\NuGet\NuGet.Config`
- `C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config`
- etc

Which might say like

1. nuget.org -> `C:\Users\<Name>\AppData\Roaming\NuGet\NuGet.Config`
2. dotnet5 ->  `C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config`
3. etc

**Comment by eerhardt** [comment]
> You would like to see the location the source originated from on disk?

Correct. I'd like to know where the source is coming from - so I can either delete it, or figure out how it got there. Is a config file on disk the only way a source can get added to my context?

In my specific case I am seeing a different set of sources between the VS dialog and what is being used on the command line. So I'd like to know where the sources on the command line are coming from.

**Comment by JonDouglas** [comment]
@eerhardt This is an excellent feature request. We'll keep this open to count 👍 & consider it for future `dotnet` releases. Thanks for the contribution.

**Comment by Frulfump** [comment]
Just ran into this issue and it would indeed be very nice to see.
For my own repos I always add a `nuget.config` with the content
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
    <clear />
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
  </packageSources>
</configuration>
```
But a freshly cloned repo didn't contain any `nuget.config` and I was trying to figure out what sources were used then and where they came from.

This issue seems to be a dupe of #9342 which was created first but this one contains more details, has more upvotes but they differ in the Priority label. This one is `Priority:3` and the older issue is `Priority:2`. So maybe the older issue should actually be closed as a duplicate?

**Comment by baronfel** [comment]
@Frulfump good call-out - in the meantime you can run `dotnet nuget config paths` to get an ordered list of all the NuGet.config files that a given workspace takes into account when computing the overall NuGet config.

**Comment by Frulfump** [comment]
Thanks! That's nice, I was looking at the help for `dotnet nuget list source` instead.
